### DISCUSSION TITLE
Add support for Content-Encoding

### DISCUSSION THREAD

**Discussion by queertypes** [falconry/falcon#2085] [Ideas]
It'd be great to facilitate the use of `Content-Encoding` and `Accepts-Encoding` request/response headers (`accepts_encoding`, `content_encoding`).

I might tackle this soon, so interacting with compressed, streaming content through Falcon is easy.

More info: [(wiki) HTTP Compression](http://en.wikipedia.org/wiki/HTTP_compression#Client.2FServer_compression_scheme_negotiation)

**Comment by lichray** [comment]
Just looked around https://gist.github.com/nvie/2769669 https://pypi.python.org/pypi/wsgigzip/0.1
It seems that WSGI server can also support this, like CherryPy's `tools.gzip.on = True`

**Comment by kgriffs** [comment]
Seems like this would be better left to the WSGI container or reverse proxy in front of your app...

**Comment by mevinbabuc** [comment]
@lichray Hey were you able to get this working on falcon itself ?

**Comment by lichray** [comment]
> @lichray Hey were you able to get this working on falcon itself ?

I haven't been using falcon for a long time.

**Comment by vytas7** [comment]
@mevinbabuc In general, it is something that is normally handled on the WSGI application server level or by a reverse proxy (like Nginx), particularly when in comes to the (request) input stream.

As to the response handling, I would also recommend handling this in a reverse proxy (very easy to enable compression in Nginx), although you can compress your responses if you really wish. You could take a look at @willnewton's https://github.com/willnewton/falcon-compression, or search PyPi for generic WSGI/ASGI compression middleware.

**Reply by mevinbabuc** [reply]
I use Gunicorn and that doesn't support gzip. 
I don't use Nginx on my box as AWS ALB interacts directly with Gunicorn to make this happen. 

Not sure if it's really worth spinning up an Nginx container on every box just to do gzip.

Let me check out the middleware. Writing a custom one didn't work as chrome/firefox kept reporting on the content encoding and content length issues.

**Reply by mevinbabuc** [reply]
Using the falcon-compression package worked for me.
